### PR TITLE
Fix #282 Added documentation for pre-push

### LINKED ISSUE #282: document pre-push hook
https://github.com/iterative/dvc/pull/1951 introduces pre-push hook.
Document the change

### PR DESCRIPTION
**Pull request by Naba7**
Fix #282  I have added documentation for ``pre-push``  git hook.

### CODE REVIEW

File: static/docs/commands-reference/install.md [hunk 1]
```diff
@@ -35,12 +35,23 @@ is a reminder to run `dvc commit`.
...
+versions or branches.
+ 
 ## Installed SCM hooks
 
 - Git `pre-commit` hook executes `dvc status` before `git commit` to inform the
   user about the workspace status.
 - Git `post-checkout` hook executes `dvc checkout` after `git checkout` to
   automatically synchronize the data files with the new workspace state.
+- Git `pre-push` hook executes `dvc push` before `git push` to upload files
+and directories under DVC control to remote.
```
Comment (shcheklein) [inline static/docs/commands-reference/install.md]: please, run prettier or install a pre-commit hook to enforce the style. There should be an extra space at the beginning of this line.

File: static/docs/commands-reference/install.md [hunk 2]
```diff
@@ -35,12 +35,23 @@ is a reminder to run `dvc commit`.
 The installed Git hook automates reminding the user to run either `dvc repro` or
 `dvc commit`.
 
+**Push** Sharing data across environments and making changes to remote storage
+is done by this command. This command searches for changed .dvc files and stages
+them for consultation. Using checksums it finds for existing entries in the
+remote cache. After that it uploads the cache files to the remote.
+
+The installed Git hook automates reminding `dvc push` and corresponds to the
```
Comment (shcheklein) [inline static/docs/commands-reference/install.md]: probably it's not about reminding, it's about actually executing `dvc push` when you run `git push, exactly like you described below (@pared can confirm this)

File: static/docs/commands-reference/install.md [hunk 3]
```diff
@@ -35,12 +35,23 @@ is a reminder to run `dvc commit`.
 The installed Git hook automates reminding the user to run either `dvc repro` or
 `dvc commit`.
 
+**Push** Sharing data across environments and making changes to remote storage
+is done by this command. This command searches for changed .dvc files and stages
```
Comment (shcheklein) [inline static/docs/commands-reference/install.md]: this command -> `dvc push`

Comment (shcheklein) [reply static/docs/commands-reference/install.md]: .dvc files -> DVC-files

Comment (shcheklein) [reply static/docs/commands-reference/install.md]: it's not looking for changed files, it takes all DVC-files from the current workspace

File: static/docs/commands-reference/install.md [hunk 4]
```diff
@@ -35,12 +35,23 @@ is a reminder to run `dvc commit`.
...
+them for consultation. Using checksums it finds for existing entries in the
+remote cache. After that it uploads the cache files to the remote.
```
Comment (shcheklein) [inline static/docs/commands-reference/install.md]: uploads files from cache that are missing on the remote storage

File: static/docs/commands-reference/install.md (hunk 1 above)
Comment (Naba7) [reply static/docs/commands-reference/install.md]: My bad, Alright changing this.

File: static/docs/commands-reference/install.md (hunk 2 above)
Comment (Naba7) [reply static/docs/commands-reference/install.md]: yes,corrected it.

File: static/docs/commands-reference/install.md (hunk 3 above)
Comment (Naba7) [reply static/docs/commands-reference/install.md]: ok,corrected.

File: static/docs/commands-reference/install.md [hunk 5]
```diff
@@ -258,3 +270,7 @@ other files but we must now commit some files to the Git repository. Looking
 closely we see that `dvc status` is again run, informing us that the data files
 are synchronized with the statement: _Pipeline is up to date. Nothing to
 reproduce_.
+
+## Example: Push for DVC
+
```
Comment (shcheklein) [inline static/docs/commands-reference/install.md]: is it empty?

File: static/docs/commands-reference/install.md [hunk 6]
```diff
@@ -35,12 +35,24 @@ is a reminder to run `dvc commit`.
 The installed Git hook automates reminding the user to run either `dvc repro` or
 `dvc commit`.
 
+**Push** Sharing data across environments and making changes to remote storage
+is done by `dvc push`. This command searches for all DVC-files from current
+workspace and stages them for consultation. Using checksums it finds for
```
Comment (shcheklein) [inline static/docs/commands-reference/install.md]: it all sounds too complicated, especially ` stages them for consultation`.

It takes all DVC-files from the current workspace, indeed. It checks if some of the data files or directories that are referenced in those DVC-files are missing on the remote storage. And it uploads them.

Let's simplify it all. `Takes DVC-files from the current workspace and saves (if they are not saved yet) files referenced in those to the remote storage.`. Something like this, please :)

File: static/docs/commands-reference/install.md (hunk 5 above)
Comment (Naba7) [reply static/docs/commands-reference/install.md]: I couldnt write any examples.

**Comment by Naba7** [comment]
I removed the example,for now.

**Pull request merged**